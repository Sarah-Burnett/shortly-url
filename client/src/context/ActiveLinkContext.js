import React, { useState, createContext } from 'react';

export const ActiveLinkContext = createContext();

export const ActiveLinkProvider = (props) => {
    const [ activeLink, setActiveLink] = useState({
        activeLink: [],
        process: false
    });

    return (
        <ActiveLinkContext.Provider value={[activeLink, setActiveLink]}>
            {props.children}
        </ActiveLinkContext.Provider>
    )
};