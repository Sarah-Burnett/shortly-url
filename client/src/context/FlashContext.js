import React, { useState, createContext } from 'react';

export const FlashContext = createContext();

export const FlashProvider = (props) => {
    const [ flash, setFlash ] = useState({
        flashMsg: "",
        isFlashVisible: false,
        isAutoHideFlash: true,
    });
    return (
        <FlashContext.Provider value={[flash, setFlash]}>
            {props.children}
        </FlashContext.Provider>
    )
};