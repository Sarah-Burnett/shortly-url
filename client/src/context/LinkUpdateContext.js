import React, { useState, createContext } from 'react';

export const LinkUpdateContext = createContext();

export const LinkUpdateProvider = (props) => {
    const [ linkUpdate, setLinkUpdate] = useState();
    return (
        <LinkUpdateContext.Provider value={[linkUpdate, setLinkUpdate]}>
            {props.children}
        </LinkUpdateContext.Provider>
    )
};