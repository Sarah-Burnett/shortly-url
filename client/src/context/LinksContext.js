import React, { useState, createContext } from 'react';

export const LinksContext = createContext();

export const LinksProvider = (props) => {
    const [ links, setLinks] = useState([]);

    return (
        <LinksContext.Provider value={[links, setLinks]}>
            {props.children}
        </LinksContext.Provider>
    )
};