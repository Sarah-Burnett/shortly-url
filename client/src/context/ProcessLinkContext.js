import React, { useState, createContext } from 'react';

export const ProcessLinkContext = createContext();

export const ProcessLinkProvider = (props) => {
    const [ processLink, setProcessLink] = useState({
        link: [],
        process: false
    });
    return (
        <ProcessLinkContext.Provider value={[processLink, setProcessLink]}>
            {props.children}
        </ProcessLinkContext.Provider>
    )
};