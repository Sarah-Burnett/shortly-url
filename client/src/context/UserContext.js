import React, { useState, createContext } from 'react';

export const UserContext = createContext();

export const UserProvider = (props) => {
    const [ user, setUser] = useState({
        isAuth: false,
        firstName: undefined,
        id: undefined,
        token: undefined,
    });
    
    return (
        <UserContext.Provider value={[user, setUser]}>
            {props.children}
        </UserContext.Provider>
    )
};