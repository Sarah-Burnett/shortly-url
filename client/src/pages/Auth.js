import React, { useState, useContext } from 'react';
import styled from 'styled-components';
import { white, breakpoint } from '../Styles';
import { Login } from '../components/Login';
import { Logout } from '../components/Logout';
import { Signup } from '../components/Signup';
import MediaQuery from 'react-responsive';
import image from '../images/illustration-working.svg';

const AuthPg = styled.div`
    min-height: 70vh;
    display: flex;
    background: ${white};
`

const Img = styled.div`
    min-height: 70vh;
    width: 70vw;
    display: flex;
    align-items: center;
    justify-content: center;
`


export const Auth = ({auth}) => {
    return (
        <AuthPg>
            <MediaQuery minDeviceWidth={breakpoint}>
                <Img><img src={image} alt="woman working at desk"/></Img>
            </MediaQuery>
            { auth==="login" && <Login/> }
            { auth==="logout" && <Logout/> }
            { auth==="signup" && <Signup/> }
        </AuthPg>
    )
}