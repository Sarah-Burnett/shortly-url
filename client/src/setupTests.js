import React from 'react';
import ReactDOM from 'react-dom';
import '@testing-library/jest-dom/extend-expect';
import { render, cleanup, fireEvent, screen, waitForElement } from '@testing-library/react';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import renderer from 'react-test-renderer';
import { UserProvider } from './context/UserContext';
import { LinksProvider } from './context/LinksContext';
import { ActiveLinkProvider } from './context/ActiveLinkContext';
import { ProcessLinkProvider } from './context/ProcessLinkContext';
import { FlashProvider } from './context/FlashContext';
import App from './App';

export const ComponentWrapper = (Component) => {
    return (
        <UserProvider>
            <LinksProvider>
                <ActiveLinkProvider>
                    <ProcessLinkProvider>
                        <FlashProvider>
                            <Component/>
                        </FlashProvider>
                    </ProcessLinkProvider>
                </ActiveLinkProvider>
            </LinksProvider>
        </UserProvider>
    )
}

export const rendersWithoutCrashing = (Component) => {
    const div = document.createElement('div');
    ReactDOM.render(<Component/>, div);
}

export const matchesSnapshot = (Component) => {
    const tree = renderer.create(<Component/>).toJSON();
    expect(tree).toMatchSnapshot();
}

export const sampleLinks = [
    {
        "date":"2020-08-14T17:29:53.041Z",
        "click":[
            {"date":1597426302926},
            {"date":1597426306071},
            {"date":1597426310231},
            {"date":1597426314763},
            {"date":1597428206995},
            {"date":1597428210093},
            {"date":1597428215131},
            {"date":1597428218755}
        ],
        "_id":"K1pjMCK-F",
        "full":"http://reactcommunity.org/react-transition-group/css-transition",
        "userId":"5ebe7d235e4a501ee8699058",
        "short":"octopus","__v":8
    },
    {
        "date":"2020-08-14T17:33:55.748Z",
        "click":[],
        "_id":"lsncCe5RV",
        "full":"https://ba-grant.netlify.app/",
        "userId":"5ebe7d235e4a501ee8699058",
        "short":"bagrant","__v":0
    }
];
