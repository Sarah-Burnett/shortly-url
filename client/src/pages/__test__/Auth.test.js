import React from 'react';
import { rendersWithoutCrashing, matchesSnapshot } from '../../setupTests';
import Auth from '../Auth';
import { UserProvider } from '../../context/UserContext';
import { LinksProvider } from '../../context/LinksContext';
import { FlashProvider } from '../../context/FlashContext';
import { BrowserRouter as Router } from 'react-router-dom';

const AuthWrapper = (auth) => {
    return (
        <UserProvider>
            <LinksProvider>
                <FlashProvider>
                    <Router>
                        <Auth auth={auth}/>
                    </Router>
                </FlashProvider>
            </LinksProvider>
        </UserProvider>
    )
};

const LoginAuth = () => AuthWrapper("login");
const LogoutAuth = () => AuthWrapper("logout");
const SignupAuth = () => AuthWrapper("signup");

it('renders login without crashing', () => rendersWithoutCrashing(LoginAuth));
it('renders logout without crashing', () => rendersWithoutCrashing(LogoutAuth));
it('renders signup without crashing', () => rendersWithoutCrashing(SignupAuth));

it('login matches snapshot', () => matchesSnapshot(LoginAuth));
it('logout matches snapshot', () => matchesSnapshot(LogoutAuth));
it('signup matches snapshot', () => matchesSnapshot(SignupAuth));

