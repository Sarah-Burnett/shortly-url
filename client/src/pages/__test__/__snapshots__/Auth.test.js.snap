// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login matches snapshot 1`] = `
<div
  className="sc-fznyAO iEXDFt"
>
  <form
    className="sc-AxhUy dLXkxw"
    data-testid="login"
    noValidate={true}
    onSubmit={[Function]}
  >
    <h3>
      Login
    </h3>
    <label
      className="sc-Axmtr fYuxyA"
    >
      Email
       
      <input
        className="sc-AxmLO erTjTr"
        id="email"
        name="email"
        onChange={[Function]}
        required={true}
        type="email"
        value=""
      />
    </label>
    <label
      className="sc-Axmtr fYuxyA"
    >
      Password
       
      <input
        className="sc-AxmLO erTjTr"
        id="password"
        name="password"
        onChange={[Function]}
        required={true}
        type="password"
        value=""
      />
    </label>
    <button
      className="sc-AxirZ sc-AxheI cOblFm"
      type="submit"
    >
      Submit
    </button>
    <button
      className="sc-AxjAm sc-AxgMl bwgmuq"
      type="button"
    >
      Forgot password
    </button>
  </form>
  <div
    className="sc-fznKkj hMwGys"
  >
    <img
      alt="woman working at desk"
      src="illustration-working.svg"
    />
  </div>
</div>
`;

exports[`logout matches snapshot 1`] = `
<div
  className="sc-fznyAO iEXDFt"
>
  <form
    className="sc-AxhUy dLXkxw"
    data-testid="logout"
  >
    <h3>
      Are you sure you want to logout?
    </h3>
    <button
      className="sc-AxirZ sc-AxheI cOblFm"
      onClick={[Function]}
    >
      Yes
    </button>
    <button
      className="sc-AxirZ sc-AxheI cOblFm"
    >
      <a
        href="/dashboard"
        onClick={[Function]}
      >
        No
      </a>
    </button>
  </form>
  <div
    className="sc-fznKkj hMwGys"
  >
    <img
      alt="woman working at desk"
      src="illustration-working.svg"
    />
  </div>
</div>
`;

exports[`signup matches snapshot 1`] = `
<div
  className="sc-fznyAO iEXDFt"
>
  <form
    className="sc-AxhUy dLXkxw"
    data-testid="signup"
    noValidate={true}
    onSubmit={[Function]}
  >
    <h3>
      Sign Up
    </h3>
    <label
      className="sc-Axmtr fYuxyA"
    >
      First Name
       
      <input
        className="sc-AxmLO erTjTr"
        id="firstName"
        name="firstName"
        onChange={[Function]}
        required={true}
        value=""
      />
    </label>
    <label
      className="sc-Axmtr fYuxyA"
    >
      Last Name
       
      <input
        className="sc-AxmLO erTjTr"
        id="lastName"
        name="lastName"
        onChange={[Function]}
        required={true}
        value=""
      />
    </label>
    <label
      className="sc-Axmtr fYuxyA"
    >
      Email
       
      <input
        className="sc-AxmLO erTjTr"
        id="email"
        name="email"
        onChange={[Function]}
        required={true}
        type="email"
        value=""
      />
    </label>
    <label
      className="sc-Axmtr fYuxyA"
    >
      Password
       
      <input
        className="sc-AxmLO erTjTr"
        id="password"
        name="password"
        onChange={[Function]}
        pattern=""
        required={true}
        type="password"
        value=""
      />
    </label>
    <label
      className="sc-Axmtr fYuxyA"
    >
      Confirm your password
       
      <input
        className="sc-AxmLO erTjTr"
        id="password2"
        name="password2"
        onChange={[Function]}
        pattern=""
        required={true}
        type="password"
        value=""
      />
    </label>
    <button
      className="sc-AxirZ sc-AxheI cOblFm"
      type="submit"
    >
      Sign up
    </button>
  </form>
  <div
    className="sc-fznKkj hMwGys"
  >
    <img
      alt="woman working at desk"
      src="illustration-working.svg"
    />
  </div>
</div>
`;
